app/Community/Components/UserProfileMeta.php

```diff
@@ -0,0 +1,227 @@
+            // Calculate based on the user's primary playing mode.
+            $preferredMode === 'softcore' ? $softcorePoints : $hardcorePoints,
+        );
+
+        $developerStats = [];
+        // FIXME: Uses legacy roles.
+        if ($this->userMassData['ContribCount'] > 0 || $this->user->getAttribute('Permissions') >= Permissions::JuniorDeveloper) {
+            $developerStats = $this->buildDeveloperStats($this->user);
+        }
+
+        $this->calculateRecentPointsEarned($this->user);
+
+        return view('community.components.user.profile-meta', [
+            'averageCompletionPercentage' => $this->averageCompletionPercentage,
+            'averageFinishedGames' => $this->calculateAverageFinishedGames($this->userJoinedGamesAndAwards),
+            'averagePointsPerWeek' => $averagePointsPerWeek,
+            'developerStats' => $developerStats,
+            'hardcoreRankMeta' => $hardcoreRankMeta,
+            'recentPointsEarned' => $this->calculateRecentPointsEarned($this->user, $preferredMode),
+            'socialStats' => $this->buildSocialStats($this->user),
+            'softcoreRankMeta' => $softcoreRankMeta,
+            'totalHardcoreAchievements' => $this->totalHardcoreAchievements,
+            'totalSoftcoreAchievements' => $this->totalSoftcoreAchievements,
+            'userClaims' => $this->userClaims,
+            'userMassData' => $this->userMassData,
+            'username' => $this->user->User,
+        ]);
+    }
+
+    private function buildRankMetadata(
+        string $username = '',
+        int $rankType = RankType::Hardcore,
+        ?int $predefinedRank = null
+    ): array {
+        $rank = $predefinedRank ?? getUserRank($username, $rankType);
+        $numRankedUsers = countRankedUsers($rankType);
+        $rankPercent = sprintf("%1.2f", ($rank / $numRankedUsers) * 100.0);
+        $rankPercentLabel = $rank > 100 ? "(Top $rankPercent%)" : "";
+        $rankOffset = (int) (($rank - 1) / 25) * 25;
+
+        return compact(
+            'rank',
+            'numRankedUsers',
+            'rankPercent',
+            'rankPercentLabel',
+            'rankOffset',
+        );
+    }
+
+    private function buildDeveloperStats(User $user): array
+    {
+        $gameAuthoredAchievementsCount = $user->authoredAchievements()
+            ->published()
+            ->select(DB::raw('COUNT(DISTINCT GameID) as game_count'))
+            ->first()
+            ->game_count;
+
+        $totalAuthoredLeaderboards = $user->authoredLeaderboards()
+            ->select(DB::raw('COUNT(LeaderboardDef.ID) AS TotalAuthoredLeaderboards'))
+            ->value('TotalAuthoredLeaderboards');
+
+        $totalAuthoredCodeNotes = $user->authoredCodeNotes()->where('Note', '!=', '')->count();
+
+        $openTickets = null;
+        // FIXME: Uses legacy roles.
+        if ($user->getAttribute('Permissions') >= Permissions::Registered) {
+            $openTickets = array_sum(countOpenTicketsByDev($user->User));
+        }
```

Comment:
banned users can have open tickets. see Euclide. Suggest basing this on "ContribCount" instead.

Reply:
Fixed in latest.